smart-contract/assembly/contracts/deweb-interface.ts

```diff
@@ -0,0 +1,80 @@
+import { Context } from '@massalabs/massa-as-sdk';
+import {
+  _onlyOwner,
+  _setOwner,
+} from '@massalabs/sc-standards/assembly/contracts/utils/ownership-internal';
+import { ChunkPost, ChunkGet } from './serializable/Chunk';
+import { Args, u32ToBytes } from '@massalabs/as-types';
+import { _getFileChunk, _getNbChunk, _setFileChunk } from './internals/chunks';
+import { FILES_PATH_LIST } from './internals/const';
+
+/**
+ * Initializes the smart contract.
+ * Sets the contract deployer as the owner and initializes an empty file path list.
+ * @param _ - Unused parameter (required).
+ */
+export function constructor(_: StaticArray<u8>): void {
+  if (!Context.isDeployingContract()) return;
+  _setOwner(Context.caller().toString());
+  FILES_PATH_LIST.set([]);
+}
+
+/**
+ * Stores file chunks in the contract storage.
+ * Only the contract owner can call this function.
+ * @param _binaryArgs - Serialized arguments containing an array of ChunkPost objects.
+ * @throws If the chunks are invalid or if the caller is not the owner.
+ */
+export function storeFileChunks(_binaryArgs: StaticArray<u8>): void {
+  _onlyOwner();
+  const args = new Args(_binaryArgs);
+  const chunks = args
+    .nextSerializableObjectArray<ChunkPost>()
+    .expect('Invalid chunks');
+
+  for (let i = 0; i < chunks.length; i++) {
+    _setFileChunk(
+      chunks[i].filePath,
+      chunks[i].id,
+      chunks[i].data,
+      chunks[i].totalChunks,
```

Comment:
sounds weird to me to find totalchunks here. it should be handeled by the contract itself, not provided by user

Reply:
Yes I understand. We did this to handle the case the totalChunks of a file changed. For example we upload a new version of the file that is shorter. In that case we would remove the extra chunks

Reply:
yes. maybe the chunk data is missing a unique identifier, like a hash of the data that you can compare to see if there is change. If you have this, you would be able to see if some chunks belong to a previous version.
Even better, if you see a file is being overridden (detecting its a new hash) you can directly remove all the chunks with a previous hash.

Reply:
i do undersant the need but i don't think its the best solution